### PR TITLE
[refactor -> 2.x & 4.x, feat & fix: 4x only] Support missing translation

### PR DESCRIPTION
**Pull request by vicb**
/ref #13861 

/cc @vikerman @gionkunz

### CODE REVIEW

File: modules/@angular/compiler/src/i18n/i18n_html_parser.ts [hunk 1]
```diff
@@ -46,7 +48,8 @@ export class I18NHtmlParser implements HtmlParser {
     }
 
     const serializer = this._createSerializer();
-    const translationBundle = TranslationBundle.load(this._translations, url, serializer);
+    const translationBundle =
+        TranslationBundle.load(this._translations, url, serializer, this._missingTranslation);
```
Comment (vicb) [inline modules/@angular/compiler/src/i18n/i18n_html_parser.ts]: +console

File: modules/@angular/compiler/src/i18n/translation_bundle.ts [hunk 2]
```diff
@@ -6,13 +6,17 @@
...
+
 import * as html from '../ml_parser/ast';
 import {HtmlParser} from '../ml_parser/html_parser';
+import {Console} from '../private_import_core';
 
 import * as i18n from './i18n_ast';
 import {I18nError} from './parse_util';
 import {PlaceholderMapper, Serializer} from './serializers/serializer';
 
+
```
Comment (vicb) [inline modules/@angular/compiler/src/i18n/translation_bundle.ts]: revert

File: modules/@angular/compiler/src/i18n/translation_bundle.ts [hunk 3]
```diff
@@ -22,16 +26,22 @@ export class TranslationBundle {
...
+      public mapperFactory?: (m: i18n.Message) => PlaceholderMapper,
+      missingTranslationStrategy: MissingTranslationStrategy = MissingTranslationStrategy.Warning,
+      console?: Console) {
+    this._i18nToHtml = new I18nToHtmlVisitor(
+        _i18nNodesByMsgId, digest, mapperFactory, missingTranslationStrategy, console);
   }
 
   // Creates a `TranslationBundle` by parsing the given `content` with the `serializer`.
-  static load(content: string, url: string, serializer: Serializer): TranslationBundle {
+  static load(
```
Comment (vicb) [inline modules/@angular/compiler/src/i18n/translation_bundle.ts]: +console

**Comment by Luftzig** [comment]
Is this already in the 2.4.8 release? Or would I have to wait until 2.4.9?

**Comment by DzmitryShylovich** [comment]
it's a feature so it's only available in 4.0

**Comment by gionkunz** [comment]
Why are features only available in 4? Did we stop pushing new features to 2.x ?

**Comment by DzmitryShylovich** [comment]
> Why are features only available in 4?

To maintain Semver. There won't be a version 2.5.x

**Comment by gionkunz** [comment]
@DzmitryShylovich You are aware that this has nothing to do with semver right? :-) While pushing breaking changes to 4.x.x you can still publish fixes and features to 2.x.x.

**Comment by DzmitryShylovich** [comment]
ok

**Comment by Maistho** [comment]
Is it possible to configure which MissingTranslationStrategy to use?

It seems to me that the AOT compiler [will always use the Warning Strategy](https://github.com/angular/angular/blob/master/packages/compiler/src/aot/compiler_factory.ts#L54).

**Comment by DzmitryShylovich** [comment]
@Maistho looks like you are right and it's not possible. please open a new issue.

**Pull request closed without merging**